### DISCUSSION TITLE
Tree is not refreshing well when I remove a node

### DISCUSSION THREAD

**Discussion by AlbaSS18** [primefaces/community#1263] [PrimeNG] [answered]
Hello,

I am using the tree component to display documents and their childs. Each document node can be edited and removed and also you can add new childs. When I filter and then, I remove an item, the tree continues displaying the node.

Below, I put images. This is my tree:
![imagen](https://github.com/primefaces/community/assets/46462037/790aad56-2fe6-4d9f-b84f-cb32e72b0c1d)

Now, I filter:
![imagen](https://github.com/primefaces/community/assets/46462037/8dbaad92-af4b-4ea6-9daf-635f3981ee4d)

Then, I delete the node with a context menu:
![imagen](https://github.com/primefaces/community/assets/46462037/a344cc0b-05a3-45ea-9578-277a176a50d5)

The node continues displaying:
![imagen](https://github.com/primefaces/community/assets/46462037/aa411649-acd7-47c6-92e0-a209328fbac8)

This is the code which I use to remove the item:
![imagen](https://github.com/primefaces/community/assets/46462037/79018c19-ba63-446a-90f8-5361fef6d6fb)

Is it a bug o am I doing something wrong?

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue?

**Reply by AlbaSS18** [reply]
Yes, here you have: https://stackblitz.com/edit/pn6pg6?file=src%2Fapp%2Fdemo%2Ftree-multiple-demo.html

Follow this steps:

1. Search `Document` in filter search.
2. Right click in parent node (Document).
3. Click in `Delete Document `option.
4. The node continues appears.

I forgot to mention that I am using version 16.

**Reply by AlbaSS18** [reply]
It looks like the problem goes much further. If I add a second child node when I am filtering and the data object has a key with array, this one is not copying.

**Reply by cetincakiroglu** [reply]
Hi,

Thank you for reporting the issue, we've opened an issue in 17.13.0 milestone. We'll try to aim this in 17.13.0 or higher versions. https://github.com/primefaces/primeng/issues/15114

**Comment by cetincakiroglu** [comment, marked as answer]
Hi @AlbaSS18,

I've reviewed the issue and found that the issue happens because you're removing the item from `value` of the tree. When you filter the nodes, value you see in the component is `filteredNodes`. In your code, you update `value` of the tree, you also need to update filteredNodes. That's why the document is removed after deleting the filter text, because you're removing the item only from `value` which is not displayed during the filter event.

Here is a working version of the stackblitz example: https://stackblitz.com/edit/pn6pg6-jcgs3h?file=src%2Fapp%2Fdemo%2Ftree-multiple-demo.ts,src%2Fapp%2Fdemo%2Ftree-multiple-demo.html

Hope it helps.
